Modify wait_for_apt function

ripesensor opened the pull request:
### What does this PR do?
Modifies the __wait_for_apt function (and others based on it) to rely on the apt exit code to know if the apt process is locked or not.

### What issues does this PR fix or reference?
#1288 

### Previous Behavior
__wait_for_apt() would do the right thing, but it was possible for another apt process to sneak in-between the function and the following apt call.

### New Behavior
Loop our apt commands until our exit code != 100.

May want to edit the comment for the function if this is a worthwhile PR.

Commit:
modify wait_for_apt function (and others based on it) to rely on the apt exit code to know if the apt process is locked or not.

vutny commented on bootstrap-salt.sh:
```diff
@@ -18,7 +18,7 @@
 set -o nounset                              # Treat unset variables as an error
 
-__ScriptVersion="2018.08.15"
+__ScriptVersion="2018.11.01"
```
Please do not bump the version number here, it would be done by a maintainer from SaltStack when the stable version will come out.

ripesensor replied:
Got it, thanks.

vutny commented on bootstrap-salt.sh:
```diff
@@ -1798,24 +1798,26 @@ __function_defined() {
-        sleep 1
-        WAIT_TIMEOUT=$((WAIT_TIMEOUT - 1))
+    # Set an initial exit code to 100; which matches the exit code of a locked apt process
+    (exit 100)
```
I think, could we execute the passed in command first, instead of simulating "locked" exit code?
This way no need to wait an additional second for each `apt` call. Although it requires to save the exit code and return it later if not equals `100`.

ripesensor replied:
I guess I was trying to not have repeated code, but it also makes more sense to at least try it instead of faking it.

vutny reviewed (changes requested):
This is great, @ripesensor !
Much cleaner implementation which is not relying on any external tool, but only on the apt utils return code.
I just have few minor comments. Thanks!

Commit:
. revert __ScriptVersion
. rework __wait_for_apt() to try our apt command first, rather
  than simulating an exit code.

Commit:
forgot to sleep, forgot to gpg sign commit

vutny commented on bootstrap-salt.sh:
```diff
@@ -1798,24 +1798,29 @@ __function_defined() {
     done
-
-    echodebug "No apt processes are currently running."
 }
```
Looks like you need to return the apt exit code in the end. It is required for the caller to be able to determine if it was not lock-related failure.
```suggestion

    return $APT_RETURN
}
```

Commit:
return correctly

vutny reviewed (approved):
Looks great!
